build-tools/packages/build-cli/test/commands/typetests.test.ts

```diff
@@ -2,11 +2,330 @@
+
+			it(`packageWithTypeValidationDisabled`, () => {
+				const pkg = clonedeep(packageWithTypeValidationDisabled);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.exist;
+				expect(pkg.typeValidation?.broken["broken-package"]).to.exist;
+				expect(pkg.typeValidation?.disabled).to.be.true;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("clearIfDisabledAndResetTrue", () => {
+			const name = "clearIfDisabledAndResetTrue";
+			const options = optionsMatrix.get(name)!;
+			it(`packageMinimal`, () => {
+				const pkg = clonedeep(packageMinimal);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.undefined;
+				expect(pkg.typeValidation?.disabled).to.be.undefined;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+
+			it(`packageWithTypeValidation`, () => {
+				const pkg = clonedeep(packageWithTypeValidation);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.empty;
+				expect(pkg.typeValidation?.disabled).to.be.false;
+				expect(pkg.devDependencies).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.4.0.0",
+				);
+			});
+
+			it(`packageWithTypeValidationDisabled`, () => {
+				const pkg = clonedeep(packageWithTypeValidationDisabled);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.empty;
+				expect(pkg.typeValidation?.disabled).to.be.true;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("clearIfDisabledAndResetFalse", () => {
+			const name = "clearIfDisabledAndResetFalse";
+			const options = optionsMatrix.get(name)!;
+			it(`packageMinimal`, () => {
+				const pkg = clonedeep(packageMinimal);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.undefined;
+				expect(pkg.typeValidation?.disabled).to.be.undefined;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+
+			it(`packageWithTypeValidation`, () => {
+				const pkg = clonedeep(packageWithTypeValidation);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.exist;
+				expect(pkg.typeValidation?.broken["broken-package"]).to.exist;
+				expect(pkg.typeValidation?.disabled).to.be.false;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.4.0.0",
+				);
+			});
```

Comment:
When checking several aspects of some produced object in a test, its often cleaner to read, has nicer errors, and easier to maintain if you just do a assert.deepEqual to a literal instead of checking all the fields seperatly.

Reply:
Good idea.

Reply:
I ended up not using this in the simplified tests, but I'll remember this for next time.